src/client/lib/ClientSocket/ClientSocket.ts

```diff
@@ -7,103 +7,173 @@ import { EventEmitter } from "events";
-
-    on (event: "chatConversationRemoved", listener: (data: { conversation: PartialChatConversation }) => void): this;
-
-    on (event: "chatMemberAdded", listener: (data: { conversation: PartialChatConversation }) => void): this;
-
-    on (event: "chatMemberLeft", listener: (data: { conversation: PartialChatConversation }) => void): this;
-
-    on (event: "friendLost", listener: (data: { user: PartialUser }) => void): this;
-
-    on (event: "friendRequest", listener: (data: { user: PartialUser }) => void): this;
-
-    on (event: "friendAdded", listener: (data: { user: PartialUser }) => void): this;
-
-    on (event: "userPresenceChanged", listener: (data: { user: PartialUser }) => void): this;
+  on(
+    event: "chatMessageSent",
+    listener: (data: { conversation: PartialChatConversation }) => void
+  ): this;
+
+  on(
+    event: "chatUserTyping",
+    listener: (data: {
+      user: PartialUser;
+      conversation: PartialChatConversation;
+    }) => void
+  ): this;
+
+  on(
+    event: "chatUserTypingStopped",
+    listener: (data: {
+      user: PartialUser;
+      conversation: PartialChatConversation;
+    }) => void
+  ): this;
+
+  on(
+    event: "chatMessage",
+    listener: (data: {
+      user: PartialUser;
+      conversation: PartialChatConversation;
+    }) => void
+  ): this;
+
+  on(
+    event: "chatMessageSent",
+    listener: (data: {
+      user: PartialUser;
+      conversation: PartialChatConversation;
+    }) => void
+  ): this;
+
+  on(
+    event: "chatConversationAdded",
+    listener: (data: { conversation: PartialChatConversation }) => void
+  ): this;
+
+  on(
+    event: "chatConversationRemoved",
+    listener: (data: { conversation: PartialChatConversation }) => void
+  ): this;
+
+  on(
+    event: "chatMemberAdded",
+    listener: (data: { conversation: PartialChatConversation }) => void
+  ): this;
+
+  on(
+    event: "chatMemberLeft",
+    listener: (data: { conversation: PartialChatConversation }) => void
+  ): this;
+
+  on(
+    event: "friendLost",
+    listener: (data: { user: PartialUser }) => void
+  ): this;
+
+  on(
+    event: "friendRequest",
+    listener: (data: { user: PartialUser }) => void
+  ): this;
+
+  on(
+    event: "friendAdded",
+    listener: (data: { user: PartialUser }) => void
+  ): this;
+
+  on(
+    event: "userPresenceChanged",
+    listener: (data: { user: PartialUser }) => void
+  ): this;
 }
 
-
 export class Socket extends EventEmitter {
-    public client: Client;
-    public socket: any;
-    public connected: boolean;
-
-    constructor (client: Client) {
-        super();
-
-        this.client = client;
-        this.socket = null;
-        this.connected = false;
-    }
-
-    connect (): Promise<void> {
-        return new Promise((resolve, reject) => {
-            const connectSocket = (retries = 0): void => {
-                this.socket = new SignalR.client("wss://realtime.roblox.com/notifications", ["usernotificationhub"], 3, true);
```

Comment:
Consider replacing usage of `signalr-client` (deprecated) with https://www.npmjs.com/package/signalr

Reply:
Do you perhaps know if Roblox uses ASP.NET or ASP.NET Core? Since Microsoft offers a signalr client as well (https://www.npmjs.com/package/@microsoft/signalr) which targets ASP.NET Core (and I'm not entirely sure if it also works with ASP.NET SignalR servers)

Reply:
> Do you perhaps know if Roblox uses ASP.NET or ASP.NET Core? Since Microsoft offers a signalr client as well ([npmjs.com/package/@microsoft/signalr](https://www.npmjs.com/package/@microsoft/signalr)) which targets ASP.NET Core (and I'm not entirely sure if it also works with ASP.NET SignalR servers)

Fairly certain `@microsoft/signalr` only works with ASP.NET Core SignalR servers, and `signalr` only works with Normal ASP.NET SignalR servers. Both `signalr` and `@microsoft/signalr` are published by the same person on NPM -- wouldn't make much sense to publish a redundant package. 

Roblox uses `ASP.NET` on `realtime.roblox.com`, judging by the headers. So `signalr` is the proper package to use. ![](https://i.jaydensar.net/2022/01/680dd6b0-1881-4635-870f-c6a74195ed78.png)

Reply:
> 

Roblox uses both SignalR for .NET Core and .NET Framework. But for the likes of Messaging Service (Message Router Service) and Real-Time notifications, which are both part of the BEDEV1 Framework (Primarily .NET Framework, becoming obsolete), they use Microsoft’s Owin integration of SignalR